app/views/stories/feeds/show.json.jbuilder

```diff
@@ -24,6 +24,11 @@ json.array!(@stories) do |article|
   end
 
   json.tag_list article.cached_tag_list_array
-  json.top_comments article.top_comments
   json.extract! article, *article_methods_to_include
+
+  json.top_comments article.top_comments do |comment|
+    json.comment_id comment.id
+    json.extract! comment, :body_markdown, :updated_at, :user_id
+    json.extract! comment.user, :username, :name
```

Comment:
Hmmmm is this going to cause n+1?

Probably fine if we only have a handful of "top comments" coming through, but the most efficient thing at query time might be to store the user info on the object like we do with article for most efficient memory use... Again probably not critical to get the interface working properly for client.

Reply:
Funny you should ask: I went back and checked and there is an N+1 with `comments`, but **only** when I have field filtering in place for the `comments` model. Taking that out (as I've done in my last commit), everything loads in 3 quarries: 

```
18:06:07 web.1       |   Article Load (0.6ms)  SELECT  "articles"."path", "articles"."title", "articles"."id", "articles"."published", "articles"."comments_count", "articles"."positive_reactions_count", "articles"."cached_tag_list", "articles"."main_image", "articles"."main_image_background_hex_color", "articles"."updated_at", "articles"."slug", "articles"."video", "articles"."user_id", "articles"."organization_id", "articles"."video_source_url", "articles"."video_code", "articles"."video_thumbnail_url", "articles"."video_closed_caption_track_url", "articles"."language", "articles"."experience_level_rating", "articles"."experience_level_rating_distribution", "articles"."cached_user", "articles"."cached_organization", "articles"."published_at", "articles"."crossposted_at", "articles"."boost_states", "articles"."description", "articles"."reading_time", "articles"."video_duration_in_seconds", "articles"."last_comment_at" FROM "articles" WHERE "articles"."published" = $1 AND (score > 7 OR featured = TRUE) ORDER BY hotness_score DESC LIMIT $2 OFFSET $3  [["published", true], ["LIMIT", 35], ["OFFSET", 1]]
18:06:07 web.1       |   Comment Load (0.4ms)  SELECT "comments".* FROM "comments" WHERE "comments"."commentable_type" = $1 AND (comments.score > 10 AND ancestry IS NULL and hidden_by_commentable_user is FALSE and deleted is FALSE) AND "comments"."commentable_id" IN ($2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14, $15, $16, $17, $18, $19, $20, $21, $22, $23, $24, $25) ORDER BY comments.score DESC  [["commentable_type", "Article"], ["commentable_id", 18], ["commentable_id", 6], ["commentable_id", 10], ["commentable_id", 16], ["commentable_id", 2], ["commentable_id", 9], ["commentable_id", 11], ["commentable_id", 12], ["commentable_id", 13], ["commentable_id", 7], ["commentable_id", 1], ["commentable_id", 15], ["commentable_id", 20], ["commentable_id", 19], ["commentable_id", 22], ["commentable_id", 25], ["commentable_id", 23], ["commentable_id", 5], ["commentable_id", 8], ["commentable_id", 24], ["commentable_id", 3], ["commentable_id", 21], ["commentable_id", 4], ["commentable_id", 17]]
18:06:07 web.1       |   User Load (0.5ms)  SELECT "users".* FROM "users" WHERE "users"."id" IN ($1, $2, $3, $4, $5, $6, $7, $8)  [["id", 5], ["id", 4], ["id", 9], ["id", 11], ["id", 1], ["id", 6], ["id", 3], ["id", 8]]
```

I'm not 100% sure which fields I'll need off of `user` right now: depending on the design from @ludwiczakpawel I may need to include more. I still think doing a larger query like this is more efficient than hoiking up fields to `comments`

Reply:
🔥